sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -780,26 +775,26 @@ public void finishBundle(Context context) throws Exception {
                 Collections.<String, String>emptyMap(),
                 requestBody,
                 new BasicHeader("", ""));
-        JsonObject searchResult = parseResponse(response);
-        boolean errors = searchResult.getAsJsonPrimitive("errors").getAsBoolean();
+        JsonNode searchResult = parseResponse(response);
+        boolean errors = searchResult.path("errors").asBoolean();
         if (errors) {
           StringBuilder errorMessages =
               new StringBuilder(
                   "Error writing to Elasticsearch, some elements could not be inserted:");
-          JsonArray items = searchResult.getAsJsonArray("items");
+          JsonNode items = searchResult.path("items");
           //some items present in bulk might have errors, concatenate error messages
-          for (JsonElement item : items) {
-            JsonObject creationObject = item.getAsJsonObject().getAsJsonObject("create");
-            JsonObject error = creationObject.getAsJsonObject("error");
+          for (JsonNode item : items) {
+            JsonNode creationObject = item.path("create");
+            JsonNode error = creationObject.get("error");
             if (error != null) {
-              String type = error.getAsJsonPrimitive("type").getAsString();
-              String reason = error.getAsJsonPrimitive("reason").getAsString();
-              String docId = creationObject.getAsJsonPrimitive("_id").getAsString();
+              String type = error.path("type").asText();
+              String reason = error.path("reason").asText();
+              String docId = creationObject.path("_id").asText();
               errorMessages.append(String.format("%nDocument id %s: %s (%s)", docId, reason, type));
-              JsonObject causedBy = error.getAsJsonObject("caused_by");
+              JsonNode causedBy = error.path("caused_by");
```

Comment:
should be `JsonNode causedBy = error.get("caused_by");` because caused_by can be null.
In the case of a missing caused_by, the output message will be 
`Caused by:  ()` instead of no caused_by output message